apps/gk_species_radiation.c

```diff
@@ -178,32 +219,30 @@ gk_species_radiation_moms(gkyl_gyrokinetic_app *app, const struct gk_species *sp
   // compute needed Maxwellian moments (n, u_par, T/m) (Jacobian factors already eliminated)
   gk_species_moment_calc(&rad->prim_moms, species->local, app->local, species->f);
 
+  gkyl_array_set_offset(rad->m0, 1.0, rad->prim_moms.marr, 0*app->confBasis.num_basis);
   gkyl_array_set_offset(rad->vtsq, 1.0, rad->prim_moms.marr, 2*app->confBasis.num_basis);
-
   gkyl_array_clear(rad->nvnu_surf, 0.0);
   gkyl_array_clear(rad->nvnu, 0.0);
   gkyl_array_clear(rad->nvsqnu_surf, 0.0);
   gkyl_array_clear(rad->nvsqnu, 0.0);
+
   for (int i=0; i<rad->num_cross_collisions; ++i) {
     // compute needed moments
     if (rad->is_neut_species[i])
-      gk_neut_species_moment_calc(&rad->moms[i], rad->collide_with_neut[i]->local, app->local, fin_neut[rad->collide_with_idx[i]]);
+      gk_neut_species_moment_calc(&rad->moms[i], rad->collide_with_neut[i]->local,
+        app->local, fin_neut[rad->collide_with_idx[i]]);
     else
       gk_species_moment_calc(&rad->moms[i], rad->collide_with[i]->local, app->local, fin[rad->collide_with_idx[i]]);
-    
     // divide out Jacobian from ion density before computation of final drag coefficient
     gkyl_dg_div_op_range(rad->moms[i].mem_geo, app->confBasis, 0, rad->moms[i].marr, 0,
       rad->moms[i].marr, 0, app->gk_geom->jacobgeo, &app->local);
-
-    const double* vtsq_min_normalized_d = gkyl_array_cfetch(rad->vtsq_min_normalized, i);    
-    gkyl_dg_calc_gk_rad_vars_nI_nu_advance(rad->calc_gk_rad_vars[i], 
-      &app->local, &species->local, 
-      rad->vnu_surf[i], rad->vnu[i], 
-      rad->vsqnu_surf[i], rad->vsqnu[i], 
-      rad->moms[i].marr, 
-      rad->nvnu_surf, rad->nvnu, 
-      rad->nvsqnu_surf, rad->nvsqnu,
-      vtsq_min_normalized_d[0], rad->vtsq);
+    gkyl_array_set_offset(rad->vtsq_min_per_species, 1.0, rad->vtsq_min_normalized, i);
```

Comment:
This and line 282 below seem avoidable because these things don't change in time/

Maybe instead of 
```
gkyl_array *vtsq_min_per_species;
```
and 
```
vtsq_min_per_species = mkarr(use_gpu, num_coll, num_dens)
```
it'd be better to create an array of arrays here, with
```
gkyl_array **vtsq_min_per_species;
vtsq_min_per_species = gkyl_malloc(numcoll*sizeof(struct gkyl_array *));
for i = 1, num_dens {
  vtsq_min_per_species[i] = mkarr(use_gpu, num_dens, 1)
}
```
and you'll have to reorganize the gkyl_array_copy and/or some memcpy so that you populate `vtsq_min_per_species` one species at a time. This way you can change avoid the `gkyl_array_set_offset(rad->vtsq_min_per_species` every time, and L245 and L291 would use `rad->vtsq_min_per_species[i]` instead.